### PR TITLE
fix string escaping

### PR DESCRIPTION
**Pull request by vbmithr**

### CODE REVIEW

**Comment by darioteixeira** [comment]
I've merged this PR and the other two. Thanks and sorry for the delay... ;-)

File: src/PGOCaml_generic.ml [hunk 1]
```diff
@@ -1569,35 +1569,13 @@ let option_map f = function
...
-    let cc = Char.code c in
-    if cc < 0x20 || cc > 0x7e then
-      Buffer.add_string buf (sprintf "\\%03o" cc) (* non-print -> \ooo *)
-    else if c = '\\' then
-      Buffer.add_string buf "\\\\" (* \ -> \\ *)
-    else
-      Buffer.add_char buf c
-  done;
-  Buffer.contents buf
+let string_of_bytea bytea = "\\x" ^ Digest.(to_hex (string bytea))
```
Comment (vouillon) [inline src/PGOCaml_generic.ml]: This looks wrong: this encodes a hash of the string and not the string itself.

Comment (jrochel) [inline src/PGOCaml_generic.ml]: Hello there! This is very wrong. `Digest.string` computes a cryptographic hash and does not belong here. If removed the problem that remains is the fact that `Digest.to_hex` works only for strings with exactly 16 characters. So we need to employ a version of `Digest.to_hex` without that precondition. I'm going work out a patch that contains a copy of `Digest.to_hex`.

File: src/PGOCaml_generic.ml [hunk 2]
```diff
@@ -1569,35 +1569,13 @@ let option_map f = function
...
-      | '"' | '\\' as x -> Buffer.add_char buf '\\'; Buffer.add_char buf x
-      | x -> Buffer.add_char buf x
-  done;
-  Buffer.contents buf
-
 let string_of_bool_array a = string_of_any_array (List.map (option_map string_of_bool) a)
 let string_of_int32_array a = string_of_any_array (List.map (option_map Int32.to_string) a)
 let string_of_int64_array a = string_of_any_array (List.map (option_map Int64.to_string) a)
-let string_of_string_array a = string_of_any_array (List.map (option_map escape_string) a)
+let string_of_string_array a = string_of_any_array (List.map (option_map String.escaped) a)
```
Comment (vouillon) [inline src/PGOCaml_generic.ml]: Are you sure Postgresql and OCaml agree on escaping?
For instance, I get this:
```
bs=# select '{"to\nto"}' :: text[];
  text   
---------
 {tonto}
```

File: src/PGOCaml_generic.ml (hunk 1 above)
Comment (jrochel) [inline src/PGOCaml_generic.ml]: https://github.com/darioteixeira/pgocaml/pull/41

**Comment by vbmithr** [comment]
Oops. Two different issues in my PR. Let me see.

**Comment by darioteixeira** [comment]
My fault, I merged the PR without taking a close look at the code.

Anyway, @vbmithr, I reckon it behooves you to provide a fix or sign off on @jrochel PR #41... ;-)

**Pull request merged**